agent/discovery/query_fetcher_v2.go

```diff
@@ -283,6 +287,50 @@ func getEndpointWeight(endpoint *pbcatalog.Endpoint, configCtx *v2DataFetcherDyn
 	return weight, true
 }
 
+// getResultPorts conditionally returns ports from a map based on a query. The results are sorted by name.
+func getResultPorts(req *QueryPayload, workloadPorts map[string]*pbcatalog.WorkloadPort) []Port {
+	if workloadPorts == nil {
+		return nil
+	}
+
+	// If a port is specified, make sure the workload implements that port name.
+	if _, ok := workloadPorts[req.PortName]; req.PortName != "" && !ok {
```

Comment:
[nit] this could go in the if below and then just becomes `if _, ok := workloadPorts[req.PortName]; !ok { ...`